### DISCUSSION TITLE
How to set in Nx monorepo tool

### DISCUSSION THREAD

**Discussion by markky21** [RyanClementsHax/storybook-addon-next#39] [Q&A] [answered]
Hello!

Is your feature request related to a problem? Please describe
I was not able to set plugin in Nx tool. Plugin require to have next.config.js file in root folder..

Describe the solution you'd like
It would be awesome if there was possibility to set plugin in monorepo managed by Nx tool.
E.g. add parameter to next.conf.js

Describe alternatives you've considered
E.g. add parameter to next.conf.js

**Comment by RyanClementsHax** [comment, marked as answer]
Thanks for starting this discussion! I'd be happy to add this as a config field. Can you open an issue and perhaps create a sample repo that we can use as a reference?

**Reply by RyanClementsHax** [reply]
I created a pull request (#40) that adds this feature. I'm just looking for an example repo to make sure I solved it properly. Are you able to provide one?

**Reply by markky21** [reply]
Hello Rayan, thx for quick responding. Yes, I will check it and also create sample repo for you. Should it be done today

**Reply by markky21** [reply]
I created PR with nx sample env.

**Reply by RyanClementsHax** [reply]
Thanks a bunch for contributing! I merged it in and was able to get the `nextConfigPath` option working (see [options](https://github.com/RyanClementsHax/storybook-addon-next/pull/41)). Let me know if it ends up not working for you.

**Reply by markky21** [reply]
Hey! Yep, the error is gone, thx! 
I was thinking about setting up storybook direct in app so ( as I understand addon features ) addon could preset next pages in the storybook..
